src/monitoring_tests/fee_test.py

```diff
@@ -0,0 +1,160 @@
+        """
+        # get settlement and trades via api
+        transaction = self.web3_api.get_transaction(tx_hash)
+        if transaction is None:
+            return False
+        settlement = self.web3_api.get_settlement(transaction)
+        trades = self.web3_api.get_trades(settlement)
+
+        partially_fillable_indices = []
+        for i, trade in enumerate(trades):
+            if trade.data.is_partially_fillable:
+                partially_fillable_indices.append(i)
+                log_output = (
+                    "Partially fillable order found.\t"
+                    + "Tx hash:"
+                    + tx_hash
+                    + "\t\t"
+                    + "Order uid: "
+                    + "NOT IMPLEMENTED YET"
+                )
+                self.logger.debug(log_output)
+
+        if len(partially_fillable_indices) > 0:
+            # get additional data for the batch
+            receipt = self.web3_api.get_receipt(tx_hash)
+            if receipt is None:
+                return False
+
+            # get batch costs
+            gas_amount, gas_price = self.web3_api.get_batch_gas_costs(
+                transaction, receipt
+            )
+
+            for i in partially_fillable_indices:
+                trade = trades[i]
+                quote = self.orderbook_api.get_quote(trade)
+                if quote is None:
+                    self.logger.error("Error fetching quote. Skipping hash %s", tx_hash)
+                    return False
+
+                gas_price_quote = self.web3_api.get_current_gas_price()
+                if gas_price_quote is None:
+                    self.logger.error(
+                        "Error fetching current gas price. Skipping hash %s", tx_hash
+                    )
+                    return False
+
+                quote.adapt_execution_to_gas_price(gas_price_quote, gas_price)
+
+                fee_amount = trade.execution.fee_amount
+                quote_fee_amount = quote.execution.fee_amount
+
+                diff_fee_abs = fee_amount - quote_fee_amount
+                diff_fee_rel = (fee_amount - quote_fee_amount) / quote_fee_amount
+
+                log_output = (
+                    "Quote test:"
+                    "Tx hash: "
+                    + tx_hash
+                    + "\t\t"
+                    + "Order indices: "
+                    + str(i)
+                    + "\t\t"
+                    + "Winning Solver: "
+                    + transaction["from"]
+                    + "\t\t"
+                    + "Fee: "
+                    + str(fee_amount)
+                    + "\t\t"
+                    + "Fee quote: "
+                    + str(quote_fee_amount)
+                    + "\t\t"
+                    + "Absolute difference: "
+                    + str(diff_fee_abs)
+                    + "\t\t"
+                    + "Relative difference: "
+                    + (str(format(100 * diff_fee_rel, ".2f")) + "%")
```

Comment:
nit: I think this multi line logging is ABSOLUTELY HIDEOUS and would strongly encourage you to use f-strings here. They may still wind up being multiline strings, but WAY less ugly and WAY more human readable. Let me see if I can take a stab at converting this one.

```py
log_output = [
    f"Quote test\nTx hash: {tx_hash}",
    f"Order indices: {i}",
    f"Winning Solver: {transaction['from']}",
    f"Fee: {fee_amount}",
    f"Fee quote: {quote_fee_amount}",
    f"Absolute difference: {diff_fee_abs}",
    f"Relative difference: {100 * diff_fee_rel:.2f}%",
]

then you can recover the string you had here with `"\t\t".join(log_output)`




You can play with it a bit here:

```py
tx_hash = "0x12"
i = "1"
transaction = {"from": "0x34"}
fee_amount = 1234
quote_fee_amount = 5678
diff_fee_abs = 32
diff_fee_rel = 0.012345

log_output = [
    f"Quote test\nTx hash: {tx_hash}",
    f"Order indices: {i}",
    f"Winning Solver: {transaction['from']}",
    f"Fee: {fee_amount}",
    f"Fee quote: {quote_fee_amount}",
    f"Absolute difference: {diff_fee_abs}",
    f"Relative difference: {100 * diff_fee_rel:.2f}%",
]

if __name__ == "__main__":
    print("\t\t".join(log_output))
```


Notice also the formated percentage string  (cf [f-string decimal formating](https://stackoverflow.com/a/45310389)).